uiServices/src/testSuiteServices_test.go

```diff
@@ -0,0 +1,187 @@
+package services
+
+import (
+	"github.com/go-chi/chi"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+)
+
+const validTestSuite = `
+{
+  "name": "TestSuiteService",
+  "testStrategy": "SuiteBased",
+  "testCases": [
+    {
+      "name":"file1.xml",
+      "preThinkTime": 1000,
+      "postThinkTime": 2000,
+      "execWeight": "Infrequent"
+    },
+    {
+      "name":"file2.xml"
+    }
+  ]
+}
+`
+const TestSuiteMissingRequired = `
+{
+  "testCases": [
+    {
+      "name":"file1.xml",
+      "preThinkTime": 1000,
+      "postThinkTime": 2000,
+      "execWeight": "Infrequent"
+    }
+  ]
+}
+`
+const TestSuiteNoName = `
+{
+  "name": "",
+  "testStrategy": "SuiteBased",
+  "testCases": [
+    {
+      "name":"file1.xml",
+      "preThinkTime": "xxxx"
+      "postThinkTime": 2000,
+      "execWeight": 123
+    }
+  ]
+}
+`
+
+func TestValidTestSuitePost(t *testing.T) {
+	r := chi.NewRouter()
+	r.Mount("/", GetIndexPage())
+
+	reader := strings.NewReader(validTestSuite)
+	r.HandleFunc("/test-suites", postTestSuites)
+
+	os.Remove(os.Getenv("GOPATH") + "/src/github.com/xtracdev/automated-perf-test/uiServices/test/TestSuiteService.xml")
+
+	filePath := os.Getenv("GOPATH") + "/src/github.com/xtracdev/automated-perf-test/uiServices/test"
+	request, err := http.NewRequest(http.MethodPost, "/test-suites", reader)
+	request.Header.Set("testSuitePathDir", filePath)
+
+	w := httptest.NewRecorder()
+	r.ServeHTTP(w, request)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	assert.Equal(t, w.Code, http.StatusCreated, "Error: Did Not Successfully Post")
+}
+
+func TestFileExistsPost(t *testing.T) {
+	r := chi.NewRouter()
+	r.Mount("/", GetIndexPage())
+
+	reader := strings.NewReader(validTestSuite)
+	r.HandleFunc("/test-suites", postTestSuites)
+
+	filePath := os.Getenv("GOPATH") + "/src/github.com/xtracdev/automated-perf-test/uiServices/test"
+	request, err := http.NewRequest(http.MethodPost, "/test-suites", reader)
+	request.Header.Set("testSuitePathDir", filePath)
+
+	w := httptest.NewRecorder()
+	r.ServeHTTP(w, request)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	assert.Equal(t, w.Code, http.StatusBadRequest, "Created, but should not have. File Should Alreday Exist")
```

Comment:
you should look at all these asserts and flip w.Code, http.StatusBadRequest so the (expected) and (actual) log correctly:

> 
> 	Error Trace:	testSuiteServices_test.go:78
> 	Error:		Not equal: 400 (expected)
> 			        != 201 (actual)
> 	Messages:	Error: Did Not Successfully Post